include/seqan3/io/structure_file/format_vienna.hpp

```diff
@@ -216,42 +216,46 @@ class structure_file_input_format_REMOVEME<format_vienna>
         }
 
         // READ STRUCTURE (if present)
+        [[maybe_unused]] size_t structure_length{};
         if constexpr (!detail::decays_to_ignore_v<structure_type>)
         {
             if constexpr (structured_seq_combined)
             {
                 assert(std::addressof(seq) == std::addressof(structure));
                 using alph_type = typename value_type_t<structure_type>::structure_alphabet_type;
-                std::ranges::copy(read_structure<alph_type>(stream_view), begin(structure));
+                // We need the structure_length parameter to count the length of the structure while reading
+                // because we cannot infer it from the (already resized) structure_seq object.
+                std::ranges::copy(read_structure<alph_type>(stream_view, structure_length), begin(structure));
 
                 if constexpr (!detail::decays_to_ignore_v<bpp_type>)
                     detail::bpp_from_rna_structure<alph_type>(bpp, structure);
             }
             else
             {
                 using alph_type = value_type_t<structure_type>;
-                std::ranges::copy(read_structure<alph_type>(stream_view), std::back_inserter(structure));
+                std::ranges::copy(read_structure<alph_type>(stream_view, structure_length),
+                                  std::back_inserter(structure));
 
                 if constexpr (!detail::decays_to_ignore_v<bpp_type>)
                     detail::bpp_from_rna_structure<alph_type>(bpp, structure);
             }
-            if constexpr (!detail::decays_to_ignore_v<seq_type>)
-                if (size(seq) != size(structure))
-                    throw parse_error{"Found sequence and associated structure of different length."};
         }
         else if constexpr (!detail::decays_to_ignore_v<bpp_type>)
         {
-            detail::bpp_from_rna_structure<wuss51>(bpp, read_structure<wuss51>(stream_view));
-
-            if constexpr (!detail::decays_to_ignore_v<seq_type>)
-                if (size(seq) != size(bpp))
-                    throw parse_error{"Found sequence and associated structure of different length."};
+            detail::bpp_from_rna_structure<wuss51>(bpp, read_structure<wuss51>(stream_view, structure_length));
         }
         else
         {
             detail::consume(stream_view | views::take_until(is_space)); // until whitespace
         }
 
+        if constexpr (!detail::decays_to_ignore_v<seq_type> &&
+                      !(detail::decays_to_ignore_v<structure_type> && detail::decays_to_ignore_v<bpp_type>))
+        {
+            if (size(seq) != structure_length)
```

Comment:
std::ranges::size? And is sized range required for seq?

Reply:
no. I changed it to distance